### DISCUSSION TITLE
Execute shows error on Next.js

### DISCUSSION THREAD

**Discussion by TidharHayute** [planetscale/discussion#558] [Q&A] [answered]
Hey there,
I'm using the planetscale-js library on my next.js project,

I run this code:

```
 const config = {
    host: process.env.NEXT_PUBLIC_PS_HOST,
    username: process.env.NEXT_PUBLIC_PS_USER,
    password: process.env.NEXT_PUBLIC_PS_PASSWORD,
  };

async function getKeyTokens() {
    try {
      const conn = await connect(config);
      console.log(conn);

      const keysList= await conn.execute(`select * from keys`);

      console.log(keysList.rows); 

      } catch (error) {
      console.log(error);
    }
}
```

And this is the error I received:
```
{
    "session": null,
    "config": {
        "host": --,
        "username": --,
        "password": --
    },
    "url": "https://aws.connect.psdb.cloud/"
}

DatabaseError
    at postJSON (index.js:148:1)
    at async Connection.execute (index.js:86:1)
    at async getKeyTokens (keys.tsx:78:26)
```

I'm not sure why I recieve this error and what went wrong, can it be related to the "session" being null?

**Comment by orware** [comment, marked as answer]
Hello @TidharHayute,

That does seem a bit odd.

How are you passing in the host value exactly? Are you passing it in simply as `aws.connect.psdb.cloud` or as a complete URL like what's shown above (e.g. `https://aws.connect.psdb.cloud/`)?

At first glance if the session isn't establishing itself correctly that would seem to indicate an issue with one of the `config` values.

Assuming that the `host`, `username`, and `password` values are all correct and accurate then the next type of test I would be curious about is if you can execute the example query below successfully that we include in our sample connection information:

```
const results = await conn.execute('select 1 from dual where 1=?', [1])
```

Or you can try running the `conn.refresh()` method after the `const conn = await connect(config);` line since that would be a manual way of establishing the session to see if that can occur without even having to run any queries ([Reference](https://github.com/planetscale/database-js/blob/main/src/index.ts#L208)). 

From what I recall, I believe the session is established lazily when your first query gets executed, rather than immediately on the `connect(config)` call so running `conn.refresh()` would allow for you to confirm if your connection details are correct and you can establish the session successfully.

**Reply by TidharHayute** [reply]
Hey there, thank you for the answer. 
It seems to be working now after what you suggested, and now I see a "session" when I run console.log(conn).

Thanks!
